src/main/java/org/commcare/cases/util/StorageBackedTreeRoot.java

```diff
@@ -146,8 +147,13 @@ private void collectNativePredicateProfiles(Vector<XPathExpression> predicates,
                             //sure the best way to do that....? Maybe tell the evaluation context to skip out here if it detects a request
                             //to resolve in a certain area?
                             Object o = FunctionUtils.unpack(((XPathEqExpr)xpe).b.eval(evalContext));
-                            optimizations.addElement(new IndexedValueLookup(filterIndex, o));
-
+                            PredicateProfile lookup;
+                            if (((XPathEqExpr)xpe).op == XPathEqExpr.EQ) {
+                                lookup = new IndexedValueLookup(filterIndex, o);
+                            } else {
```

Comment:
This should be a positive match for `== NEQ`, and if that fails it should either throw a not-implemented exception or (more likely) should just fail to trigger an optimization, otherwise any new `eq` expression type will be optimized as a negative indexed value lookup.